torch/distributed/_shard/sharded_tensor/interface.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+import abc
+
+
+import torch
+
+import torch.distributed._shard.sharding_spec as shard_spec
+
+from .shard import Shard
+from .utils import _flatten_tensor_size
+
+
+class ShardedTensorInterface(torch.Tensor):
+    @staticmethod
+    def __new__(cls,
+                sharding_spec: shard_spec.ShardingSpec,
+                *size,
+                **kwargs):
+        # Use __new__ for logging purposes.
+        torch._C._log_api_usage_once("torch.distributed._shard.sharded_tensor")
+        sizes = _flatten_tensor_size(size)
+        dtype = kwargs['dtype']
+        layout = kwargs['layout']
+        pin_memory = kwargs['pin_memory']
+        requires_grad = kwargs['requires_grad']
+        r = torch.Tensor._make_wrapper_subclass(
+            cls,
+            sizes,
+            dtype=dtype,
+            layout=layout,
+            pin_memory=pin_memory,
+            requires_grad=requires_grad
+        )
+        return r
```

Comment:
`r` is referencing to the unpartitioned tensor, that would be sharded according to the spec? Also, would it be possible to pass a tensor to the constructor and create a ShardedTensorInterface -- or it's not needed?

Reply:
If `r` doesn't have any memory attached (which is desirable), do we implicitly collect and show the unpartitioned, `r` in this case, when accessed? Also, is there a variable pointing to the individual shards/handles, or it's up to the child implementation to deifne?

Reply:
r is merely a wrapper tensor with no memory attached as you can see, you can define both global tensor or local shards in the subclass. When accessing the instance of the ShardedTensor interface in python, it will be directed to `__torch_function__` (for python only ops like `__get__`, and to `__torch_dispatch__`, where you can control how the access should behave(i.e. return attributes of the full tensors, operating on individual shards, etc.)